app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -565,6 +579,19 @@ else if(resultCode == RESULT_OK){
     }
 
     /**
+     * @return If the nature of the data that the session is waiting for has not changed since the
+     * callout that we are returning from was made
+     */
+    private boolean sessionStateUnchangedSinceCallout(CommCareSession session, Intent intent) {
+        boolean dataIdsEqual = session.getNeededData().equals(
+                intent.getStringExtra(KEY_PENDING_SESSION_DATA));
+        String intentDatum = intent.getStringExtra(KEY_PENDING_SESSION_DATUM_ID);
+        boolean datumIdsEqual = intentDatum  == null ? true :
+                intentDatum.equals(session.getNeededDatum().getDataId());
+        return dataIdsEqual && datumIdsEqual;
```

Comment:
These seem very close, it took me two reads to realize they weren't the same value